### PR TITLE
docs: draft proposal for docs improvement 

### PR DESCRIPTION
**Pull request by dhmlau**
Based on @raymondfeng's draft https://github.com/strongloop/loopback-next/pull/2925/, the purpose of this PR is to create a concrete plan on what we should do at a high level and for each section. 

The goal is that we can create smaller tasks out of this proposal.

### CODE REVIEW

**Comment by dhmlau** [comment]
@strongloop/loopback-maintainers, I'd like to get your feedback on how we categorize our docs at a higher level first and the naming.  Thanks.

File: docs/site/site-proposal.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+docs improvement, I'd like to propose the following changes.
+
+## High-level sidebar
+
+- **Overview**: No change. Just add a dedicated "overview" item on the sidebar.
+  It still points to [index.md](index.md) file.
+- **Getting Started**: no change. [Getting-started.md](Getting-started.md)
+- **Tutorials**: [Tutorials.md](Tutorials.md).
+
+- **Basic constructs**: Same as
```
Comment (dhmlau) [inline docs/site/site-proposal.md]: maybe keeping `Key Concepts`

File: docs/site/site-proposal.md [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+  - Sending
+  - Rejecting
+- Interceptors
+
+**Extending LoopBack**
+
+- Extension point/extension
+- Discovering and ordering
+- Chain of handling
+
```
Comment (bajtos) [inline docs/site/site-proposal.md]: At the moment, we have the following sections in `Extending LoopBack`:

Screen Shot 2019-06-24 at 09 27 26

What's the plan - are we going to preserve them unchanged?

From my point of view, I feel it's important to provide good documentation for extension developers, showing them best practices for building npm modules providing reusable functionality for LB4 applications.

Comment (dhmlau) [reply docs/site/site-proposal.md]: I've added them under "Extending LoopBack". Thanks.

File: docs/site/site-proposal.md [hunk 3]
```diff
@@ -164,6 +164,69 @@ docs improvement, I'd like to propose the following changes.
...
+- API docs
+- For LoopBack 3.x users
+  - Differences between v3 and v4
+  - Migration guide
+- Best Practices
+- References
+
+Question: I have trouble mapping the following contents. Please help:
+
+- Serving static files ??
```
Comment (raymondfeng) [inline docs/site/site-proposal.md]: This should go under `Usage scenario`.

File: docs/site/site-proposal.md [hunk 4]
```diff
@@ -164,6 +164,69 @@ docs improvement, I'd like to propose the following changes.
...
+- Self-hosted REST API Explorer -> Under "Extending LoopBack"?- Boot and Mount a
```
Comment (raymondfeng) [inline docs/site/site-proposal.md]: This should go under `Create APIs`.

File: docs/site/site-proposal.md [hunk 5]
```diff
@@ -164,6 +164,69 @@ docs improvement, I'd like to propose the following changes.
...
+  LoopBack 3 Application -> Under "For LoopBack 3.x users"?
+- Remove "Crafting LoopBack 4"??
```
Comment (raymondfeng) [inline docs/site/site-proposal.md]: We can have an `Architecture` section, including `Behind the scene`

Comment (dhmlau) [reply docs/site/site-proposal.md]: I've renamed the `behind the scene` to `architecture` now.

**Comment by dhmlau** [comment]
Closing this PR as some has turned into GH issues, and another round of improvement is led by Miroslav.

**Pull request closed without merging**